src/libs/lib.h

```diff
@@ -64,6 +64,16 @@ typedef struct dt_lib_t
       void (*set_sample_point)(struct dt_lib_module_t *self, float x, float y);
     } colorpicker;
 
+    /** Histogram processing hooks */
+    struct
+    {
+      struct dt_lib_module_t *module;
+      void (*process)(struct dt_lib_module_t *self, const void *const input,
+                      int width, int height, int stride, gboolean is_8bit);
+      // FIXME: should this be a function or just a boolean which histogram lib keeps updated?
+      // FIXME: should this be a darktable-level value, set by lib/histogram.c and noticed by iops?
+      gboolean is_linear;
```

Comment:
I've some worry about this -- read/write isn't protected by any sort of mutex. I think the worst case would be that an iop would be redrawing its histogram when a keyboard accelerator changed the histogram scale, and the iop's histogram might redraw with the wrong scale. But I don't have a deep understanding of multi-threading in darktable and all the risks/safeties needed.

Reply:
It is not protected but cannot be broken as a gboolean is read/written in an atomic operation anyway. So I suppose this is ok.